pkg/istio/authorizationpolicy/controller.go

```diff
@@ -0,0 +1,456 @@
+func (c *Controller) resync(stopCh <-chan struct{}) {
+	t := time.NewTicker(c.apResyncInterval)
+	defer t.Stop()
+	for {
+		select {
+		case <-t.C:
+			log.Infoln("Running resync for authorization policies...")
+			apListRaw := c.authzpolicyIndexInformer.GetIndexer().List()
+			for _, adRaw := range apListRaw {
+				c.ProcessConfigChange(model.EventAdd, adRaw)
+			}
+		case <-stopCh:
+			log.Infoln("Stopping authorization policies resync...")
+			return
+		}
+	}
+}
+
+// checkAuthzEnabledAnnotation checks if current servce object has "authz.istio.io/enabled" annotation set
+func (c *Controller) checkAuthzEnabledAnnotation(serviceObj *corev1.Service) bool {
+	if _, ok := serviceObj.Annotations[authzEnabledAnnotation]; ok {
+		if serviceObj.Annotations[authzEnabledAnnotation] == authzEnabled {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *Controller) createDryrunResource(convertedCR model.Config, authzPolicyName string, namespace string) error {
+	convertedObj, err := crd.ConvertConfig(collections.IstioSecurityV1Beta1Authorizationpolicies, convertedCR)
+	if err != nil {
+		return fmt.Errorf("unable to convert authorization policy config to istio objects, resource name: %v", convertedCR.Name)
+	}
+	configInBytes, err := yaml.Marshal(convertedObj)
+	if err != nil {
+		return fmt.Errorf("could not marshal %v: %v", convertedCR.Name, err)
+	}
+	yamlFileName := authzPolicyName + "--" + namespace + ".yaml"
+	return ioutil.WriteFile(DryRunStoredFilesDirectory + yamlFileName, configInBytes, 0644)
+}
+
+func (c *Controller) findDeleteDryrunResource(authzPolicyName string, namespace string) error {
+	yamlFileName := authzPolicyName + "--" + namespace + ".yaml"
+	if _, err := os.Stat(DryRunStoredFilesDirectory + yamlFileName); os.IsNotExist(err) {
+		log.Infof("file %s does not exist in local directory\n", DryRunStoredFilesDirectory + yamlFileName)
+		return nil
+	}
+	log.Infof("deleting file under path: %s\n", DryRunStoredFilesDirectory + yamlFileName)
+	return os.Remove(DryRunStoredFilesDirectory + yamlFileName)
+}
+
+func (c *Controller) createAuthzPolicyResource(convertedCR model.Config, obj *corev1.Service) error {
+	// form the authorization policy config and send create sign to the queue
+	athenzDomainRaw, exists, err := c.adIndexInformer.GetIndexer().GetByKey(athenz.NamespaceToDomain(obj.Namespace))
+	if err != nil {
+		return fmt.Errorf("error when getting athenz domain from cache: %v", err.Error())
+	}
+	if !exists {
+		// processing controller to delete them
+		return fmt.Errorf("athenz domain %v does not exist in cache", obj.Namespace)
+	}
+
+	athenzDomain, ok := athenzDomainRaw.(*adv1.AthenzDomain)
+	if !ok {
+		return errors.New("athenz domain cast failed")
+	}
+	signedDomain := athenzDomain.Spec.SignedDomain
+	labels := obj.GetLabels()
+	domainRBAC := athenz.ConvertAthenzPoliciesIntoRbacModel(signedDomain.Domain, &c.adIndexInformer)
+	if _, ok := labels["svc"]; !ok {
```

Comment:
Should we move the labels check under `labels := obj.GetLabels()`? 

This code block
```domainRBAC := athenz.ConvertAthenzPoliciesIntoRbacModel(signedDomain.Domain, &c.adIndexInformer)
convertedCR = c.rbacProvider.ConvertAthenzModelIntoIstioAuthzPolicy(domainRBAC, obj.Namespace, obj.Name, labels["svc"])
log.Infoln("Creating Authz Policy ... ")
if !c.dryrun {
	revision, err := c.configStoreCache.Create(convertedCR)
	if err != nil {
		log.Errorln("error creating authz policy:", err.Error())
		return err
	}
	log.Debugln("Created revision number is: ", revision)
} else {
	err := c.createDryrunResource(convertedCR, obj.Name, obj.Namespace)
	if err != nil {
		return fmt.Errorf("unable write to file, err: %v", err)
	}
}
```
seems to be used multiple times for create/update in processSvcResource/processAthenzDomainResource/processAuthorizationPolicyResource. Can we make a function and reuse it? We can add if statements inside the function to check for create/update/delete operations maybe?

Reply:
I think for events, we can keep it separate for better readability. I updated conversion into a help function to reduce code duplication